src/drafty.js

```diff
@@ -2452,12 +2495,15 @@ function shortenTree(tree, limit, tail) {
       node.text = tail;
       limit = -1;
     } else if (node.text) {
-      const len = node.text.length;
-      if (len > limit) {
-        node.text = node.text.substring(0, limit) + tail;
+      const graphemes = getGraphemesFromString(node.text);
+      if (graphemes.length > limit) {
+        node.text = graphemes
+          .slice(0, limit)
+          .map((segment) => segment.segment)
+          .join("") + tail;
```

Comment:
Code style: please use `'` (single quotes) for strings which require no localization.